### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by lukastaegert**
## Rollup Plugin Name: `*`

This PR contains:

- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [x] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

### Description
This updates all dependencies for all plugins, using one commit per plugin. At the moment, there are some security issues that prevent PRs from passing CI.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -33,15 +33,6 @@ module.exports = {
...
-      'error',
-      {
-        arrowParens: 'always',
-        printWidth: 100,
-        singleQuote: true,
-        trailingComma: 'none',
-        plugins: ['prettier-plugin-package']
-      }
-    ]
+    'prettier/prettier': 'error'
```
Comment (lukastaegert) [inline .eslintrc.js:36]: I have moved prettier config to a dedicated file that is shared in all places and provides better IDE support. Relevant change is probably that the `printWidth` in README files is now 100 characters as well. If this is not desired, I can add an override for markdown files to set it to 80 again.

File: packages/alias/rollup.config.js [hunk 2]
```diff
@@ -6,8 +6,8 @@ export default {
   input: 'src/index.ts',
   external: [...Object.keys(pkg.dependencies), 'os'],
   output: [
-    { file: pkg.main, format: 'cjs' },
+    { file: pkg.main, format: 'cjs', exports: 'auto' },
```
Comment (lukastaegert) [inline packages/alias/rollup.config.js:9]: These changes are to silence a warning in latest Rollup.

File: packages/alias/rollup.config.js [hunk 3]
```diff
@@ -6,8 +6,8 @@ export default {
...
     { file: pkg.module, format: 'es' }
   ],
-  plugins: [typescript()]
+  plugins: [typescript({ sourceMap: false })]
```
Comment (lukastaegert) [inline packages/alias/rollup.config.js:12]: Apparently the TypeScript plugin complains if I do not generate a sourcemap unless this is set to false. I assume this will also improve performance.

File: packages/auto-install/package.json [hunk 4]
```diff
@@ -11,16 +11,17 @@
...
   "main": "dist/index.js",
+  "module": "dist/index.es.js",
   "scripts": {
     "build": "rollup -c",
     "ci:coverage": "nyc pnpm run test && nyc report --reporter=text-lcov > coverage.lcov",
     "ci:lint": "pnpm run build && pnpm run lint",
     "ci:lint:commits": "commitlint --from=${CIRCLE_BRANCH} --to=${CIRCLE_SHA1}",
     "ci:test": "pnpm run test -- --verbose",
     "lint": "pnpm run lint:js && pnpm run lint:docs && pnpm run lint:package",
-    "lint:docs": "prettier --single-quote --write README.md",
```
Comment (lukastaegert) [inline packages/auto-install/package.json:21]: These options are now handled via the `.prettierrc` file

**Comment by lukastaegert** [comment]
This is now ready for review

**Comment by shellscape** [comment]
@lukastaegert I'll take a look at this tomorrow, was out in the Gulf of Mexico all day today and I'm whooped. Thanks for doing this.

**Comment by lukastaegert** [comment]
Sure, no problem, I envy you! There are some changes I made that you should think about if it is ok, mainly the prettier change. On the other hand, I do not think there is a need to release new versions of any plugins just because of this.

File: .eslintignore [hunk 5]
```diff
@@ -1 +1 @@
-!.eslintrc.js
+.eslintrc.js
```
Comment (shellscape) [inline .eslintignore:1]: any reason this was changed? we'd like ESLint to ensure that the config file matches style as well.

File: .eslintrc.js (hunk 1 above)
Comment (shellscape) [reply .eslintrc.js:36]: Yeah please do revert this. This configuration keeps prettier and ESLint from fighting, and requires only a single editor extension. There's a lot of debate about this, but reverting this change is going to cause more trouble than it's worth.

File: packages/auto-install/rollup.config.js [hunk 6]
```diff
@@ -4,10 +4,10 @@ import pkg from './package.json';
 
 export default {
   input: 'src/index.ts',
-  plugins: [typescript()],
+  plugins: [typescript({ sourceMap: false })],
```
Comment (shellscape) [inline packages/auto-install/rollup.config.js:7]: why are we removing the sourcemaps?

File: packages/beep/package.json [hunk 7]
```diff
@@ -39,10 +39,13 @@
...
   },
   "devDependencies": {
-    "rollup": "^2.0.0",
-    "strip-ansi": "^5.2.0"
+    "rollup": "^2.23.0",
+    "strip-ansi": "^6.0.0"
   },
   "ava": {
+    "babel": {
+      "compileEnhancements": false
```
Comment (shellscape) [inline packages/beep/package.json:47]: can we document (or mention here) why this change was needed now and not previously?

File: packages/dynamic-import-vars/rollup.config.js [hunk 8]
```diff
@@ -4,7 +4,7 @@ export default {
   input: 'src/index.js',
   external: [...Object.keys(pkg.dependencies), 'path'],
   output: [
-    { file: pkg.main, format: 'cjs' },
+    { file: pkg.main, format: 'cjs', exports: 'named' },
```
Comment (shellscape) [inline packages/dynamic-import-vars/rollup.config.js:7]: is this due to new rollup features?

File: packages/yaml/rollup.config.js [hunk 9]
```diff
@@ -1,26 +1,10 @@
-import babel from 'rollup-plugin-babel';
-
 import pkg from './package.json';
 
 export default {
   input: 'src/index.js',
-  plugins: [
```
Comment (shellscape) [inline packages/yaml/rollup.config.js:7]: what's the reason for removing the babel plugin?

File: packages/pluginutils/package.json [hunk 10]
```diff
@@ -79,13 +82,11 @@
...
-  "module": "./dist/es/index.js",
   "nyc": {
     "extension": [
       ".js",
       ".ts"
     ]
   },
-  "type": "commonjs",
-  "types": "types/index.d.ts"
+  "type": "commonjs"
```
Comment (shellscape) [inline packages/pluginutils/package.json:91]: could you explain this property? I'm not finding it in the typescript docs https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

**Comment by lukastaegert** [comment]
> Definitely want to revert the prettier changes.

There is more to it than just reverting because some changes need to be done. Basically what prettier 2 does is it changes a few defaults that do not fit our current code-style. If we do not want a .prettierrc file, then we need to amend each call to the prettier CLI in every single package.json file with a `--trailing-comma none` and possibly an option for `--arrow-parens` as well, making this less and less maintainable.

Also it does not use a chance we have with the mono-repo to enforce consistent styling with shared configuration that is easily updated in a central place.

I added another commit to show-case that this is also very flexible by enforcing the previous print width of 80 for README files only. I can of course merge this into the previous commits again, this is just for demonstration:

```js
{
  "arrowParens": "always",
  "printWidth": 100,
  "singleQuote": true,
  "trailingComma": "none",
  "plugins": ["prettier-plugin-package"],
  "overrides": [
    {
      "files": "README.md",
      "options": {
        "arrowParens": "avoid",
        "printWidth": 80
      }
    }
  ]
}
```

This will also be picked up by ESLint but also has better tooling support so that you can have e.g. prettier on save instead of having to trigger eslint manually time and again. While I agree this is strictly not necessary, I would urge you to use this chance since I needed to touch everything anyway to reduce maintenance overhead here and improve consistency.

File: .prettierrc [hunk 11]
```diff
@@ -3,5 +3,14 @@
   "printWidth": 100,
   "singleQuote": true,
   "trailingComma": "none",
-  "plugins": ["prettier-plugin-package"]
+  "plugins": ["prettier-plugin-package"],
+  "overrides": [
```
Comment (shellscape) [inline .prettierrc:7]: I think I'm missing some context on this addition.

File: .eslintrc.js (hunk 1 above)
Comment (lukastaegert) [reply .eslintrc.js:36]: And if we just double the config from the .prettierrc file in the .eslintrc.js file? In any case, I would really love to get rid of the manual prettier `--print-width` params in favour of central config. As the eslintrc is central as well, that would be fine with me.

File: .prettierrc (hunk 11 above)
Comment (lukastaegert) [reply .prettierrc:7]: Before, `README` files were prettiered with the default `printWidth` of 80 and the Prettier 1 default of avoiding arrow parantheses. That way, this config applies to `package.json` files with the old values. But I would be happy to align Readme files with the remaining formatting.

**Comment by shellscape** [comment]
> prettier on save instead of having to trigger eslint manually time and again

That's the part that confuses me. Atom, VSCode, VIM, and other editors all have an ESLint plugin that will run automatically. It negates the need for running prettier independently - and it honestly cleans up the conflicts between ESLint and prettier rather well. Prior to merging the two, they were constantly at odds. That's why we went that direction. I recently introduced the same change at work and we've got ~50 devs working the same way daily, with a big improvement in DX. There are different schools of thought on this, but I'd really encourage not trying to change the status quo that we've been operating on in the plugins repo for a good while.

If Prettier@2 is the source of the problem, then is there a compelling reason to update that?

In the mean time, I'll take a look at Prettier@2 and see what's up with the markdown files.

File: packages/pluginutils/package.json (hunk 10 above)
Comment (lukastaegert) [reply packages/pluginutils/package.json:91]: This is a Node 13+ property to designate that files below this package file are NOT ES modules. It is the inverse of `"type": "module"`, but it is also equivalent to not specifying it at all https://nodejs.org/api/esm.html#esm_package_json_type_field
Not sure who added it originally.

File: .prettierrc (hunk 11 above)
Comment (shellscape) [reply .prettierrc:7]: I see, thanks. (also got context from your last PR-reply).

File: packages/yaml/rollup.config.js (hunk 9 above)
Comment (lukastaegert) [reply packages/yaml/rollup.config.js:7]: As we now have a Node 10 baseline, I do not think it makes sense to explicitly transpile down to loader compatibility targets as it both bloats files and makes them less readable. You may have hit an important point here, though: Technically, this would make this a breaking change as we would lift the compatibility baseline to Node 10. I would still love to keep this change but add a "BREAKING CHANGE" note to the commits if you agree. No need to release a new versions right away, though, IMO.

File: packages/dynamic-import-vars/rollup.config.js (hunk 8 above)
Comment (lukastaegert) [reply packages/dynamic-import-vars/rollup.config.js:7]: This one is actually an old feature: If there is a default and named exports, Rollup switches to named exports mode but displays a warning. This was the case for this plugin, making it explicit silences the warning.

File: packages/pluginutils/package.json (hunk 10 above)
Comment (shellscape) [reply packages/pluginutils/package.json:91]: really appreciate the explanation. probably sounded like I was questioning it on the PR, poor choice of words. I was honestly hoping you knew.

File: packages/beep/package.json (hunk 7 above)
Comment (lukastaegert) [reply packages/beep/package.json:47]: I believe this is due to how Ava 3 handles ESM. Not sure the `compileEnhancements` are strictly necessary, though, I was copying this from other places, but according to the docs it needed to be moved into the Babel option. There is a lot of plugins that need this now, you can easily test it: If you remove the Babel option entirely, the tests will fail. Should I document it for each plugin? Otherwise, I would rather hope that people look into the Ava docs when in doubt.

File: packages/yaml/rollup.config.js (hunk 9 above)
Comment (shellscape) [reply packages/yaml/rollup.config.js:7]: Please feel free to update the engine to v10 in package.json on a new commit for the plugin (and any others that were changed with this) as a breaking change. Totally worth a major, and we've been kicking the can down the road on doing that.

File: packages/dynamic-import-vars/rollup.config.js (hunk 8 above)
Comment (shellscape) [reply packages/dynamic-import-vars/rollup.config.js:7]: Thanks!

File: packages/auto-install/rollup.config.js (hunk 6 above)
Comment (lukastaegert) [reply packages/auto-install/rollup.config.js:7]: They were never present. Rollup needs to have the sourcemap flag for them to be generated. But the new TypeScript plugins warns when the Rollup and plugin options do not match. Honestly I think it is ok for libraries to not ship sourcemaps, but of course we could consider adding them for our own testing if we want.

File: packages/beep/package.json (hunk 7 above)
Comment (shellscape) [reply packages/beep/package.json:47]: Nah. I'm familiar with the option and suspected as much but wanted to get your take. Your comment here is good enough.

**Comment by lukastaegert** [comment]
No really, it was just two relevant defaults that are different. I can add the flags to the manual calls in the package.json files.

**Comment by shellscape** [comment]
Gotcha. Before you do that, please give me a bit to see if there's another way to go about that. I have an idea I need to test.

**Comment by lukastaegert** [comment]
Ok, then I will not do that yet. Thanks for giving a little more background about why it was handled that way.

File: .eslintignore (hunk 5 above)
Comment (lukastaegert) [reply .eslintignore:1]: It was getting a "Parsing error: 'parserOptions.project' has been set to @typescript-eslint-parser'. The file does not match your project config: .eslintrc.js.
The file must be included in at least one of the projects provided."
If you have an idea to make it work, I have not.

File: packages/yaml/rollup.config.js (hunk 9 above)
Comment (lukastaegert) [reply packages/yaml/rollup.config.js:7]: I added the engines property into the original commits and added them as breaking changes to the commit messages

**Comment by lukastaegert** [comment]
I did revert the prettiert changes now, in the end it was really only one additional parameter. I hope this can be merged now. Only open point is the `eslintignore` that is still excluded due to issues with the typescript parser.

**Comment by shellscape** [comment]
Sorry for the delay. I'm on the tail end of a 15 hour day today and I'm wiped. You aren't forgotten I promise!

**Comment by lukastaegert** [comment]
No worries.
